vault/diagnose/tls_verification.go

```diff
@@ -52,23 +53,45 @@ func ListenerChecks(ctx context.Context, listeners []listenerutil.Listener) ([]s
 
 		// Perform checks on the TLS Cryptographic Information.
 		warnings, err := TLSFileChecks(l.TLSCertFile, l.TLSKeyFile)
-		for _, warning := range warnings {
-			warning = listenerID + ": " + warning
+		listenerWarnings, listenerErrors = appendWarErr(ctx, warnings, listenerWarnings, err, listenerErrors, listenerID)
+
+		// Perform checks on the Client CA Cert
+		warnings, err = TLSClientCAFileCheck(l)
+		listenerWarnings, listenerErrors = appendWarErr(ctx, warnings, listenerWarnings, err, listenerErrors, listenerID)
+
+		// TODO: Use listenerutil.TLSConfig to warn on incorrect protocol specified
+		// Alternatively, use tlsutil.SetupTLSConfig.
+	}
+	return listenerWarnings, listenerErrors
+}
+
+func appendWarErr(ctx context.Context, newWarnings, listenerWarnings []string, newErr error, listenerErrors []error, listenerID string) ([]string, []error) {
+	for _, warning := range newWarnings {
+		warning = listenerID + ": " + warning
+		if !containsString(listenerWarnings, warning) {
 			listenerWarnings = append(listenerWarnings, warning)
 			Warn(ctx, warning)
 		}
-		if err != nil {
-			errMsg := listenerID + ": " + err.Error()
-			listenerErrors = append(listenerErrors, fmt.Errorf(errMsg))
-			Error(ctx, fmt.Errorf(errMsg))
-		}
+	}
+	if newErr != nil {
+		errMsg := listenerID + ": " + newErr.Error()
+		listenerErrors = append(listenerErrors, fmt.Errorf(errMsg))
+		Error(ctx, fmt.Errorf(errMsg))
 
-		// TODO: Use listenerutil.TLSConfig to warn on incorrect protocol specified
-		// Alternatively, use tlsutil.SetupTLSConfig.
 	}
 	return listenerWarnings, listenerErrors
 }
 
+func containsString(messages []string, lookup_msg string) bool {
```

Comment:
Instead of checking for duplicates, if there are duplicates that are legitimately necessary we should strive to distinguish the warning messages.